### PR TITLE
Add VerifyIndexedAttestation for v0.6

### PR DESCRIPTION
**Pull request by 0xKiwi**
Part of #2307 

Adds [VerifyIndexedAttestation](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_beacon-chain.md#verify_indexed_attestation) to `beacon-chain/core/blocks/block_operations.go`

### CODE REVIEW

File: beacon-chain/core/blocks/block_operations.go [hunk 1]
```diff
@@ -590,6 +592,80 @@ func VerifyAttestation(beaconState *pb.BeaconState, att *pb.Attestation, verifyS
...
+	custodyBit0Indices := indexedAtt.CustodyBit_0Indices
+	custodyBit1Indices := indexedAtt.CustodyBit_1Indices
+
+	custodyBitIntersection := sliceutil.Intersection(custodyBit0Indices, custodyBit1Indices)
+	if len(custodyBitIntersection) != 0 {
+		return fmt.Errorf("custody bit indice should not contain duplicates, received: %v", custodyBitIntersection)
+	}
+
+	if len(custodyBit1Indices) > 0 {
+		return fmt.Errorf("custody bit 1 indices should be empty for phase 0")
```
Comment (terencechain) [inline beacon-chain/core/blocks/block_operations.go]: We should distinguish `return false` and `assert` like in spec.
Something that fails `assertion` invalids the block versus something that returns false does not invalid the block.

I suggest we change the return signature to `(error, bool)`, and this should be `nil, false`

File: beacon-chain/core/blocks/block_operations.go [hunk 2]
```diff
@@ -590,6 +592,80 @@ func VerifyAttestation(beaconState *pb.BeaconState, att *pb.Attestation, verifyS
...
+	}
+
+	maxIndices := params.BeaconConfig().MaxIndicesPerAttestation
+	totalIndicesLength := uint64(len(custodyBit0Indices) + len(custodyBit1Indices))
+	if maxIndices < totalIndicesLength || totalIndicesLength < 1 {
+		return fmt.Errorf("total length of indices is not proper length, received: %d",
```
Comment (terencechain) [inline beacon-chain/core/blocks/block_operations.go]: looks like this should be `nil, false` too

File: beacon-chain/core/blocks/block_operations.go [hunk 3]
```diff
@@ -590,6 +592,80 @@ func VerifyAttestation(beaconState *pb.BeaconState, att *pb.Attestation, verifyS
...
+//        signature=indexed_attestation.signature,
+//        domain=get_domain(state, DOMAIN_ATTESTATION, slot_to_epoch(indexed_attestation.data.slot)),
+//    )
+func VerifyIndexedAttestation(state *pb.BeaconState, indexedAtt *pb.IndexedAttestation) error {
+	custodyBit0Indices := indexedAtt.CustodyBit_0Indices
+	custodyBit1Indices := indexedAtt.CustodyBit_1Indices
+
+	custodyBitIntersection := sliceutil.Intersection(custodyBit0Indices, custodyBit1Indices)
+	if len(custodyBitIntersection) != 0 {
+		return fmt.Errorf("custody bit indice should not contain duplicates, received: %v", custodyBitIntersection)
```
Comment (terencechain) [inline beacon-chain/core/blocks/block_operations.go]: this is correct, triggering this error will invalid the block

File: beacon-chain/core/blocks/block_operations.go (hunk 1 above)
Comment (0xKiwi) [reply beacon-chain/core/blocks/block_operations.go]: 👍

File: beacon-chain/core/blocks/block_operations.go [hunk 4]
```diff
@@ -538,6 +541,78 @@ func VerifyAttestation(beaconState *pb.BeaconState, att *pb.Attestation, verifyS
...
+func VerifyIndexedAttestation(state *pb.BeaconState, indexedAtt *pb.IndexedAttestation) (bool, error) {
+	custodyBit0Indices := indexedAtt.CustodyBit_0Indices
+	custodyBit1Indices := indexedAtt.CustodyBit_1Indices
+
+	custodyBitIntersection := sliceutil.Intersection(custodyBit0Indices, custodyBit1Indices)
+	if len(custodyBitIntersection) != 0 {
+		return false, fmt.Errorf("custody bit indice should not contain duplicates, received: %v", custodyBitIntersection)
+	}
+
+	if len(custodyBit1Indices) > 0 {
```
Comment (prestonvanloon) [inline beacon-chain/core/blocks/block_operations.go]:
```suggestion
      // To be removed in phase 1
	if len(custodyBit1Indices) > 0 {
```

File: beacon-chain/core/blocks/block_operations.go (hunk 1 above)
Comment (prestonvanloon) [reply beacon-chain/core/blocks/block_operations.go]: I disagree with the signature, the thing you want is always first (the error is always last)

**Pull request merged**